framework/src/actions/ActionFactory.C

```diff
@@ -40,15 +42,26 @@ ActionFactory::reg(const std::string & name,
 
 std::shared_ptr<Action>
 ActionFactory::create(const std::string & action,
-                      const std::string & action_name,
+                      const std::string & full_action_name,
                       InputParameters & parameters)
 {
+  std::string action_name = MooseUtils::shortName(full_action_name);
   parameters.addPrivateParam("_moose_app", &_app);
   parameters.addPrivateParam("action_type", action);
   std::pair<ActionFactory::iterator, ActionFactory::iterator> iters;
 
+  if (!(parameters.have_parameter<bool>("isObjectAction") &&
+        parameters.get<bool>("isObjectAction")))
+    parameters.set<std::vector<std::string>>("control_tags")
+        .push_back(MooseUtils::baseName(full_action_name));
```

Comment:
Can we do this towards the end of `Parser::walkRaw` in the vicinity of where we modify `control_tags` for the object params of moose object actions? In this location it just feels out of the blue and a developer who comes along will have to go search through the code base to understand why there is special treatment given to non-moose-object-actions

Reply:
There we do not have access to the parameters in the warehouse. For moose object actions we can do that because we are modifying the parameters of the objects but not the actions. I am guessing it does not hurt to remove the if statement here.

Reply:
You are operating on the exact same parameters object here. This code in `Parser::walkRaw`:
```c++
    // Create the Action
    std::shared_ptr<Action> action_obj =
        _action_factory.create(it->second._action, MooseUtils::shortName(curr_identifier), params);
```

Reply:
right. Will do. We can add this before calling input warehouse add input param function.